src/Core/Listener/AdminGroupLockingPrevention.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Core\Listener;
+
+use Flarum\Core\Exception\PermissionDeniedException;
+use Flarum\Core\Group;
+use Flarum\Event\UserWillBeSaved;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class AdminGroupLockingPrevention
+{
+    /**
+     * @param Dispatcher $events
+     */
+    public function subscribe(Dispatcher $events)
+    {
+        $events->listen(UserWillBeSaved::class, [$this, 'whenUserWillBeSaved']);
+    }
+
+    /**
+     * @param UserWillBeSaved $event
+     * @throws PermissionDeniedException
+     */
+    public function whenUserWillBeSaved(UserWillBeSaved $event)
+    {
+        $actor = $event->actor;
+        $user = $event->user;
+        $groups = array_get($event->data, 'relationships.groups.data');
+
+        // Prevent an admin from removing their admin permission via the API
+        if ($groups && $actor->id === $user->id && $actor->isAdmin()) {
+            $adminInGroups = array_filter($groups, function ($group) {
+                return $group['id'] == Group::ADMINISTRATOR_ID;
+            });
+            $keepsAdmin = count($adminInGroups) > 0;
+
+            if (! $keepsAdmin) {
+                throw new PermissionDeniedException;
```

Comment:
I can't, unless I add a new exception & exception handler, as `PermissionDeniedExceptionHandler` only uses the status from the exception itself.